elastic-apm-agent/src/main/java/co/elastic/apm/agent/premain/ShadedClassLoader.java

```diff
@@ -157,7 +160,15 @@ private Class<?> defineClass(String name, byte[] classBytes) {
                 }
             }
         }
-        return defineClass(name, classBytes, 0, classBytes.length, ShadedClassLoader.class.getProtectionDomain());
+
+        ProtectionDomain protectionDomain = AccessController.doPrivileged(new PrivilegedAction<ProtectionDomain>() {
```

Comment:
This allocates a `PrivilegedAction` instance for every loaded class, although we only need a single `ProtectionDomain`. Let's execute this once and keep the reference for this `ProtectionDomain`.